update autoconf, automake to recent versions for gcc7-centos6 images

we have to update autoconf & automake versions on our `gcc7-centos6` images
currently, [gettext_installer](https://github.com/bincrafters/conan-gettext_installer) [fails](https://travis-ci.com/bincrafters/conan-gettext_installer/jobs/171421611) with the following error:
```
aclocal.m4:17: warning: this file was generated for autoconf 2.69.
You have another version of autoconf.  It may work, but is not guaranteed to.
If you have problems, you may need to regenerate the build system entirely.
To do so, use the procedure documented by the package, typically 'autoreconf'.
configure.ac:26: error: Autoconf version 2.65 or higher is required
aclocal.m4:156: AM_INIT_AUTOMAKE is expanded from...
configure.ac:26: the top level
autom4te: /usr/bin/m4 failed with exit status: 63
WARNING: 'autoconf' is probably too old.
         You should only need it if you modified 'configure.ac',
         or m4 files included by it.
         The 'autoconf' program is part of the GNU Autoconf package:
         <http://www.gnu.org/software/autoconf/>
         It also requires GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [Makefile:402: configure] Error 63
```
running `autoreconf -fiv` doesn't cure this issue, as project uses macro from newer automake version.
this might not be easy to do, EPEL release has [autoconf268](https://www.rpmfind.net/linux/rpm2html/search.php?query=autoconf268) package, but no recent automake. this may end up into building autoconf/automake from source, or building RPM packages.

/cc @uilianries

--- a/gcc_7-centos6-x86/Dockerfile
+++ b/gcc_7-centos6-x86/Dockerfile
@@ -27,6 +27,7 @@ RUN printf "i686" >  /etc/yum/vars/arch \
        mpfr-devel \
        libmpc-devel \
        nasm \
+       m4 \
        libffi-devel \
        pkgconfig \
        subversion \

--- a/gcc_7-centos6-x86/Dockerfile
+++ b/gcc_7-centos6-x86/Dockerfile
@@ -65,15 +65,6 @@ RUN printf "i686" >  /etc/yum/vars/arch \
     && make install \
     && popd \
     && rm -rf /tmp/automake-1.16* \
-    && wget -O /tmp/sudo-1.8.27.tar.gz --no-check-certificate --quiet https://www.sudo.ws/dist/sudo-1.8.27.tar.gz \
-    && tar -zxf /tmp/sudo-1.8.27.tar.gz -C /tmp \
-    && pushd /tmp/sudo-1.8.27 \
-    && ./configure --prefix=/usr \
-    && make -s \
-    && make install \
-    && popd \
-    && rm -rf /tmp/sudo-1.8.27* \
-    && rm /opt/rh/devtoolset-7/root/usr/bin/sudo \
     && wget --no-check-certificate --quiet -O /tmp/cmake-3.12.1.tar.gz https://cmake.org/files/v3.12/cmake-3.12.1.tar.gz \
     && tar -xzf /tmp/cmake-3.12.1.tar.gz -C /tmp \
     && pushd /tmp/cmake-3.12.1 \

--- a/gcc_7-centos6-x86/Dockerfile
+++ b/gcc_7-centos6-x86/Dockerfile
@@ -19,6 +19,7 @@ RUN printf "i686" >  /etc/yum/vars/arch \
     && printf "i386" >  /etc/yum/vars/basearch \
     && yum update -y \
     && yum install -y \
+       sudo \
        wget \
        git \
        make \

--- a/gcc_7-centos6-x86/Dockerfile
+++ b/gcc_7-centos6-x86/Dockerfile
@@ -19,7 +19,6 @@ RUN printf "i686" >  /etc/yum/vars/arch \
     && printf "i386" >  /etc/yum/vars/basearch \
     && yum update -y \
     && yum install -y \
-       sudo \
        wget \
        git \
        make \
@@ -28,7 +27,6 @@ RUN printf "i686" >  /etc/yum/vars/arch \
        mpfr-devel \
        libmpc-devel \
        nasm \
-       autoconf \
        libffi-devel \
        pkgconfig \
        subversion \
@@ -49,6 +47,32 @@ RUN printf "i686" >  /etc/yum/vars/arch \
        perl-core \
        help2man \
     && yum clean all \
+    && wget -O /tmp/autoconf-2.69.tar.gz --no-check-certificate --quiet https://ftp.gnu.org/gnu/autoconf/autoconf-2.69.tar.gz \
+    && tar -zxf /tmp/autoconf-2.69.tar.gz -C /tmp \
+    && pushd /tmp/autoconf-2.69 \
+    && ./configure --prefix=/usr \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/autoconf-2.69* \
+    && wget -O /tmp/automake-1.16.tar.gz --no-check-certificate --quiet https://ftp.gnu.org/gnu/automake/automake-1.16.tar.gz \
+    && tar -zxf /tmp/automake-1.16.tar.gz -C /tmp \
+    && pushd /tmp/automake-1.16 \
+    && ./configure --prefix=/usr \
+    && sed -i "s/'none';/'reduce';/g" bin/automake.in \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/automake-1.16* \
+    && wget -O /tmp/sudo-1.8.27.tar.gz --no-check-certificate --quiet https://www.sudo.ws/dist/sudo-1.8.27.tar.gz \
+    && tar -zxf /tmp/sudo-1.8.27.tar.gz -C /tmp \
+    && pushd /tmp/sudo-1.8.27 \
+    && ./configure --prefix=/usr \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/sudo-1.8.27* \
+    && rm /opt/rh/devtoolset-7/root/usr/bin/sudo \
     && wget --no-check-certificate --quiet -O /tmp/cmake-3.12.1.tar.gz https://cmake.org/files/v3.12/cmake-3.12.1.tar.gz \
     && tar -xzf /tmp/cmake-3.12.1.tar.gz -C /tmp \
     && pushd /tmp/cmake-3.12.1 \

--- a/gcc_7-centos6/Dockerfile
+++ b/gcc_7-centos6/Dockerfile
@@ -16,7 +16,6 @@ ENV PATH=/opt/rh/rh-python36/root/usr/bin:/opt/rh/devtoolset-7/root/usr/bin:${PA
 RUN yum update -y \
     && yum install -y centos-release-scl \
     && yum install -y \
-       sudo \
        wget \
        git \
        make \
@@ -27,7 +26,7 @@ RUN yum update -y \
        mpfr-devel \
        libmpc-devel \
        nasm \
-       autoconf \
+       m4 \
        libffi-devel \
        openssl-devel \
        pkgconfig \
@@ -41,6 +40,32 @@ RUN yum update -y \
        rh-python36 \
        help2man \
     && yum clean all \
+    && wget -O /tmp/autoconf-2.69.tar.gz --no-check-certificate --quiet https://ftp.gnu.org/gnu/autoconf/autoconf-2.69.tar.gz \
+    && tar -zxf /tmp/autoconf-2.69.tar.gz -C /tmp \
+    && pushd /tmp/autoconf-2.69 \
+    && ./configure --prefix=/usr \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/autoconf-2.69* \
+    && wget -O /tmp/automake-1.16.tar.gz --no-check-certificate --quiet https://ftp.gnu.org/gnu/automake/automake-1.16.tar.gz \
+    && tar -zxf /tmp/automake-1.16.tar.gz -C /tmp \
+    && pushd /tmp/automake-1.16 \
+    && ./configure --prefix=/usr \
+    && sed -i "s/'none';/'reduce';/g" bin/automake.in \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/automake-1.16* \
+    && wget -O /tmp/sudo-1.8.27.tar.gz --no-check-certificate --quiet https://www.sudo.ws/dist/sudo-1.8.27.tar.gz \
+    && tar -zxf /tmp/sudo-1.8.27.tar.gz -C /tmp \
+    && pushd /tmp/sudo-1.8.27 \
+    && ./configure --prefix=/usr \
+    && make -s \
+    && make install \
+    && popd \
+    && rm -rf /tmp/sudo-1.8.27* \
+    && rm /opt/rh/devtoolset-7/root/usr/bin/sudo \
     && wget -O /tmp/cmake-3.12.1-Linux-x86_64.sh --no-check-certificate --quiet 'https://cmake.org/files/v3.12/cmake-3.12.1-Linux-x86_64.sh' \
     && bash /tmp/cmake-3.12.1-Linux-x86_64.sh --prefix=/usr/local --exclude-subdir \
     && rm /tmp/cmake-3.12.1-Linux-x86_64.sh \
@@ -55,5 +80,3 @@ RUN yum update -y \
 
 USER conan
 WORKDIR /home/conan
-
-

--- a/gcc_7-centos6/Dockerfile
+++ b/gcc_7-centos6/Dockerfile
@@ -16,6 +16,7 @@ ENV PATH=/opt/rh/rh-python36/root/usr/bin:/opt/rh/devtoolset-7/root/usr/bin:${PA
 RUN yum update -y \
     && yum install -y centos-release-scl \
     && yum install -y \
+       sudo \
        wget \
        git \
        make \

--- a/gcc_7-centos6/Dockerfile
+++ b/gcc_7-centos6/Dockerfile
@@ -57,15 +57,6 @@ RUN yum update -y \
     && make install \
     && popd \
     && rm -rf /tmp/automake-1.16* \
-    && wget -O /tmp/sudo-1.8.27.tar.gz --no-check-certificate --quiet https://www.sudo.ws/dist/sudo-1.8.27.tar.gz \
-    && tar -zxf /tmp/sudo-1.8.27.tar.gz -C /tmp \
-    && pushd /tmp/sudo-1.8.27 \
-    && ./configure --prefix=/usr \
-    && make -s \
-    && make install \
-    && popd \
-    && rm -rf /tmp/sudo-1.8.27* \
-    && rm /opt/rh/devtoolset-7/root/usr/bin/sudo \
     && wget -O /tmp/cmake-3.12.1-Linux-x86_64.sh --no-check-certificate --quiet 'https://cmake.org/files/v3.12/cmake-3.12.1-Linux-x86_64.sh' \
     && bash /tmp/cmake-3.12.1-Linux-x86_64.sh --prefix=/usr/local --exclude-subdir \
     && rm /tmp/cmake-3.12.1-Linux-x86_64.sh \

